File: src/main/java/architecture/resonator_combat_framework/core/RcfClient.java
package architecture.resonator_combat_framework.core;

import net.minecraft.client.Minecraft;
import net.neoforged.api.distmarker.Dist;
import net.neoforged.bus.api.SubscribeEvent;
import net.neoforged.fml.ModContainer;
import net.neoforged.fml.common.EventBusSubscriber;
import net.neoforged.fml.common.Mod;
import net.neoforged.fml.event.lifecycle.FMLClientSetupEvent;

@Mod(value = Rcf.ID, dist = Dist.CLIENT)
@EventBusSubscriber(modid = Rcf.ID, value = Dist.CLIENT)
public class RcfClient {
	public RcfClient(ModContainer container) {
	}

	@SubscribeEvent
	static void onClientSetup(FMLClientSetupEvent event) {
		Rcf.LOGGER.info("HELLO FROM CLIENT SETUP");
		Rcf.LOGGER.info("MINECRAFT NAME >> {}", Minecraft.getInstance().getUser().getName());
	}
}

File: src/main/kotlin/architecture/resonator_combat_framework/core/RcfClient.kt
package architecture.resonator_combat_framework.core

import architecture.resonator_combat_framework.util.RcfUtil
import net.minecraft.client.Minecraft
import net.neoforged.api.distmarker.Dist
import net.neoforged.bus.api.SubscribeEvent
import net.neoforged.fml.common.EventBusSubscriber
import net.neoforged.fml.common.Mod
import net.neoforged.fml.event.lifecycle.FMLClientSetupEvent
import net.neoforged.neoforge.client.gui.ConfigurationScreen
import net.neoforged.neoforge.client.gui.IConfigScreenFactory
import thedarkcolour.kotlinforforge.neoforge.forge.LOADING_CONTEXT
import thedarkcolour.kotlinforforge.neoforge.forge.MOD_BUS

@Mod(value = RcfUtil.ID, dist = [Dist.CLIENT])
@EventBusSubscriber(modid = RcfUtil.ID, value = [Dist.CLIENT])
object RcfClient {
	init {
		val modContainer = LOADING_CONTEXT.activeContainer
		val modBus = MOD_BUS
		modContainer.registerExtensionPoint(
			IConfigScreenFactory::class.java,
			IConfigScreenFactory(::ConfigurationScreen)
		)
	}

	@SubscribeEvent
	fun onClientSetup(event: FMLClientSetupEvent) {
		RcfUtil.LOGGER.info("HELLO FROM CLIENT SETUP")
		RcfUtil.LOGGER.info("MINECRAFT NAME >> {}", Minecraft.getInstance().user.name)
	}
}
